CRM/MembershipExtras/Service/MembershipInstallmentsHandler.php

```diff
@@ -213,19 +213,25 @@ private function createLineItems(CRM_Contribute_BAO_Contribution $contribution)
         $entityID = $contribution->id;
       }
 
+
+      $taxAmount = 0;
+      if (!empty($lineItem['tax_amount'])) {
+        $taxAmount = $lineItem['tax_amount'];
+      }
+
       $lineItemParms = [
         'entity_table' => $lineItem['entity_table'],
         'entity_id' => $entityID,
         'contribution_id' => $contribution->id,
-        'price_field_id' => $lineItem['price_field_id'],
+        'price_field_id' => CRM_Utils_Array::value('price_field_id', $lineItem),
         'label' => $lineItem['label'],
         'qty' => $lineItem['qty'],
         'unit_price' => $lineItem['unit_price'],
         'line_total' => $lineItem['line_total'],
-        'price_field_value_id' => $lineItem['price_field_value_id'],
+        'price_field_value_id' => CRM_Utils_Array::value('price_field_value_id', $lineItem),
         'financial_type_id' => $lineItem['financial_type_id'],
         'non_deductible_amount' => $lineItem['non_deductible_amount'],
-        'tax_amount' => $lineItem['tax_amount'],
+        'tax_amount' => CRM_Utils_Array::value('tax_amount', $lineItem),
```

Comment:
Is there a reason why only some of `$lineItem`'s values are accessed using `CRM_Utils_Array::value()`?

Reply:
yes, if you are paying for a membership along with extra contribution, the extra contribution line ittem will not have all the data available in the membership line item such as the price_set_id since they will be  NULL for the extra contribution line item.